.github/workflows/my-ci-pipeline.yml

```diff
@@ -0,0 +1,33 @@
+name: Continuous Integration 
+on:
+  push:
+    branches: [main]
+    paths-ignore:
+      - 'README.md'
+  pull_request:
+    branches: [main]
+    paths-ignore:
+      - 'README.md'    
+
+jobs: 
+ build: 
+  name: Build and test 
+  runs-on: ubuntu-latest 
+  steps: 
+   - uses: actions/checkout@v2 
+   - run: docker build --target test --tag todo-app:test .
+   - run: docker run -e TRELLO_KEY={secrets.TRELLO_KEY} -e TRELLO_TOKEN={secrets.TRELLO_TOKEN} -e TRELLO_BOARD_ID={secrets.TRELLO_BOARD_ID} todo-app:test
+  
+ publish:
+  name: Pubish artefacts
+  needs: build
```

Comment:
We probably want to only publish our artefacts when they're on the main branch, so adding a check here to limit this while still allowing the whole workflow (and hence the tests) to run on every commit would be good, that can look something like:
```
  if: github.ref == 'refs/heads/main' && github.event_name != 'pull_request'
```